File: app/src/main/java/fudi/freddy/mokitosample/contact_view/ContactPresent.java
package fudi.freddy.mokitosample.contact_view;

import fudi.freddy.mokitosample.model.Contact;
import fudi.freddy.mokitosample.store.StoreContract;
import fudi.freddy.mokitosample.utilStore.ContractsUniversal;

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 8/13/17
 */

public class ContactPresent implements ContactContract.UserActions {

    private ContactContract.ContactView contactView;
    private StoreContract storeContract = ContractsUniversal.getStoreContract();

    public ContactPresent(ContactContract.ContactView contactView) {
        this.contactView = contactView;
    }

    @Override
    public void saveContact() {
//        storeContract.save(new Contact(
//                contactView.loadInput3(),
//                contactView.loadInput2(),
//                contactView.loadInput1(),
//                contactView.loadInput4(),
//                contactView.loadInput5()
//        ));
        storeContract.save(contactView.getContactView());
        contactView.clearText();
        contactView.display("Data guardada");
    }

    @Override
    public void loadContact(String mail) {
        contactView.display((Contact) storeContract.load(mail));
    }
}

File: app-kotlin/src/main/java/fudi/freddy/mokitosample/contact_view/ContactPresent.kt
package fudi.freddy.mokitosample.contact_view

import android.support.annotation.VisibleForTesting
import fudi.freddy.mokitosample.contact_view.ContactContract.ContactView
import fudi.freddy.mokitosample.contact_view.ContactContract.UserActions
import fudi.freddy.mokitosample.model.Contact
import fudi.freddy.mokitosample.store.StoreContract
import fudi.freddy.mokitosample.utilStore.ContractsUniversal

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 8/13/17
 */
class ContactPresent(private val contactView: ContactView) : UserActions {
    private var storeContract = ContractsUniversal.storeContract as StoreContract<Contact>

    override fun saveContact() {
        storeContract.save(contactView.contact)
        contactView.clearText()
        contactView.display("Data guardada")
    }

    override fun loadContact(mail: String?) {
        val value = storeContract.load(mail!!)
        contactView.display(value)
    }

    internal fun visibleForTesting(storeContract: StoreContract<Contact>){
        this.storeContract = storeContract
    }

}